src/crypto/tls/ech.go

```diff
@@ -0,0 +1,1093 @@
+// Copyright 2020 Cloudflare, Inc. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"errors"
+	"fmt"
+	"github.com/cloudflare/circl/hpke"
+	"io"
+
+	"golang.org/x/crypto/cryptobyte"
```

Comment:
nit: I learned long ago that you're supposed to put stdlib imports and imports from external packages in different stanzas.
```suggestion
	"io"

        "github.com/cloudflare/circl/hpke"
	"golang.org/x/crypto/cryptobyte"
```

Reply:
Thanks for flagging, I'll fix it with the next push. I ran `gofmt` and assumed it would fix the import ordering. Apparently not good enough, so I ran `goimports` on every commit to fix it:

```
git rebase -i 94b33ada97~ --exec 'cd src && $HOME/go/bin/goimports -l -w crypto/tls/*.go crypto/x509/*.go'
git diff    # after reviewing and potential manual edits:
git commit --amend -a --no-edit && git rebase --continue
```

I also checked whether I missed anything:
```
git grep -ne \"github.com/cloudflare/ -C3 src/crypto
```

The changes compared to the initial version should be pretty non-controversial:
<details>
<summary>git diff -U2 2f61bdf8df 5e7998ded1</summary>

```diff
diff --git a/src/crypto/tls/auth.go b/src/crypto/tls/auth.go
index f1ff78e322..604e431540 100644
--- a/src/crypto/tls/auth.go
+++ b/src/crypto/tls/auth.go
@@ -6,7 +6,4 @@ package tls
 
 import (
-	circlPki "github.com/cloudflare/circl/pki"
-	circlSign "github.com/cloudflare/circl/sign"
-
 	"bytes"
 	"crypto"
@@ -19,4 +16,7 @@ import (
 	"hash"
 	"io"
+
+	circlPki "github.com/cloudflare/circl/pki"
+	circlSign "github.com/cloudflare/circl/sign"
 )
 
diff --git a/src/crypto/tls/ech.go b/src/crypto/tls/ech.go
index 0c717c0008..8e71a3a8b8 100644
--- a/src/crypto/tls/ech.go
+++ b/src/crypto/tls/ech.go
@@ -7,7 +7,8 @@ import (
 	"errors"
 	"fmt"
-	"github.com/cloudflare/circl/hpke"
 	"io"
 
+	"github.com/cloudflare/circl/hpke"
+
 	"golang.org/x/crypto/cryptobyte"
 )
diff --git a/src/crypto/tls/ech_config.go b/src/crypto/tls/ech_config.go
index ec9985121f..62de3cec54 100644
--- a/src/crypto/tls/ech_config.go
+++ b/src/crypto/tls/ech_config.go
@@ -7,7 +7,8 @@ import (
 	"errors"
 	"fmt"
+	"io"
+
 	"github.com/cloudflare/circl/hpke"
 	"github.com/cloudflare/circl/kem"
-	"io"
 
 	"golang.org/x/crypto/cryptobyte"
diff --git a/src/crypto/tls/ech_provider.go b/src/crypto/tls/ech_provider.go
index 7868a812b0..d114efb276 100644
--- a/src/crypto/tls/ech_provider.go
+++ b/src/crypto/tls/ech_provider.go
@@ -7,4 +7,5 @@ import (
 	"errors"
 	"fmt"
+
 	"github.com/cloudflare/circl/hpke"
 	"github.com/cloudflare/circl/kem"
diff --git a/src/crypto/tls/generate_cert.go b/src/crypto/tls/generate_cert.go
index 237292390c..360ce8fa80 100644
--- a/src/crypto/tls/generate_cert.go
+++ b/src/crypto/tls/generate_cert.go
@@ -20,6 +20,4 @@ import (
 	"encoding/pem"
 	"flag"
-	circlSign "github.com/cloudflare/circl/sign"
-	circlSchemes "github.com/cloudflare/circl/sign/schemes"
 	"log"
 	"math/big"
@@ -28,4 +26,7 @@ import (
 	"strings"
 	"time"
+
+	circlSign "github.com/cloudflare/circl/sign"
+	circlSchemes "github.com/cloudflare/circl/sign/schemes"
 )
 
diff --git a/src/crypto/tls/generate_delegated_credential.go b/src/crypto/tls/generate_delegated_credential.go
index c93c321a02..1b72b414d5 100644
--- a/src/crypto/tls/generate_delegated_credential.go
+++ b/src/crypto/tls/generate_delegated_credential.go
@@ -24,9 +24,10 @@ import (
 	"flag"
 	"fmt"
-	circlSign "github.com/cloudflare/circl/sign"
 	"log"
 	"os"
 	"path/filepath"
 	"time"
+
+	circlSign "github.com/cloudflare/circl/sign"
 )
 
diff --git a/src/crypto/tls/handshake_server.go b/src/crypto/tls/handshake_server.go
index dcfcfb2564..3c1d219e7a 100644
--- a/src/crypto/tls/handshake_server.go
+++ b/src/crypto/tls/handshake_server.go
@@ -6,6 +6,4 @@ package tls
 
 import (
-	circlSign "github.com/cloudflare/circl/sign"
-
 	"context"
 	"crypto"
@@ -20,4 +18,6 @@ import (
 	"io"
 	"time"
+
+	circlSign "github.com/cloudflare/circl/sign"
 )
 
diff --git a/src/crypto/tls/hpke.go b/src/crypto/tls/hpke.go
index 1361d58a0b..b3861d0951 100644
--- a/src/crypto/tls/hpke.go
+++ b/src/crypto/tls/hpke.go
@@ -7,4 +7,5 @@ import (
 	"errors"
 	"fmt"
+
 	"github.com/cloudflare/circl/hpke"
 )
diff --git a/src/crypto/tls/tls.go b/src/crypto/tls/tls.go
index aaceb4978b..973aac4379 100644
--- a/src/crypto/tls/tls.go
+++ b/src/crypto/tls/tls.go
@@ -54,6 +54,4 @@ package tls
 
 import (
-	circlSign "github.com/cloudflare/circl/sign"
-
 	"bytes"
 	"context"
@@ -69,4 +67,6 @@ import (
 	"os"
 	"strings"
+
+	circlSign "github.com/cloudflare/circl/sign"
 )
 
diff --git a/src/crypto/tls/tls_cf.go b/src/crypto/tls/tls_cf.go
index 65db04e02d..c9e089fa9f 100644
--- a/src/crypto/tls/tls_cf.go
+++ b/src/crypto/tls/tls_cf.go
@@ -5,9 +5,9 @@ package tls
 
 import (
+	"time"
+
 	circlPki "github.com/cloudflare/circl/pki"
 	circlSign "github.com/cloudflare/circl/sign"
-
 	"github.com/cloudflare/circl/sign/eddilithium3"
-	"time"
 )
 
diff --git a/src/crypto/tls/tls_cf_circl_test.go b/src/crypto/tls/tls_cf_circl_test.go
index 92c577852e..e731e8b2f6 100644
--- a/src/crypto/tls/tls_cf_circl_test.go
+++ b/src/crypto/tls/tls_cf_circl_test.go
@@ -9,9 +9,10 @@ import (
 	"crypto/x509/pkix"
 	"fmt"
-	"github.com/cloudflare/circl/sign"
-	"github.com/cloudflare/circl/sign/eddilithium3"
 	"math/big"
 	"testing"
 	"time"
+
+	"github.com/cloudflare/circl/sign"
+	"github.com/cloudflare/circl/sign/eddilithium3"
 )
 
diff --git a/src/crypto/x509/parser.go b/src/crypto/x509/parser.go
index 00fb32828b..7c8117d616 100644
--- a/src/crypto/x509/parser.go
+++ b/src/crypto/x509/parser.go
@@ -26,8 +26,8 @@ import (
 	"unicode/utf8"
 
+	circlPki "github.com/cloudflare/circl/pki"
+
 	"golang.org/x/crypto/cryptobyte"
 	cryptobyte_asn1 "golang.org/x/crypto/cryptobyte/asn1"
-
-	circlPki "github.com/cloudflare/circl/pki"
 )
 
diff --git a/src/crypto/x509/x509_cf.go b/src/crypto/x509/x509_cf.go
index b597ad78f4..aab1c6480c 100644
--- a/src/crypto/x509/x509_cf.go
+++ b/src/crypto/x509/x509_cf.go
@@ -2,11 +2,10 @@ package x509
 
 import (
+	"crypto"
+	"encoding/asn1"
+
 	circlPki "github.com/cloudflare/circl/pki"
 	circlSign "github.com/cloudflare/circl/sign"
-
 	"github.com/cloudflare/circl/sign/eddilithium3"
-
-	"crypto"
-	"encoding/asn1"
 )
 
```
</details>